### PR TITLE
Update CVO to flat updatepayload structure

### PR DESCRIPTION
**Pull request by abhinavdahiya**
* updates the `lib/resourcebuilder` to be allow for adding custom resource builders
* add operatorstatus custom resource builder
* update syncs

/cc @yifan-gu

### CODE REVIEW

File: lib/resourcebuilder/interface.go [hunk 1]
```diff
@@ -2,37 +2,70 @@ package resourcebuilder
...
+var (
+	// Mapper is default ResourceMapper.
+	Mapper = NewResourceMapper()
 )
 
+// ResourceMapper maps {Group, Version} to a function that returns Interface and an error.
+type ResourceMapper struct {
+	l *sync.Mutex
+
+	gvkToNew map[schema.GroupVersionKind]NewInteraceFunc
```
Comment (yifan-gu) [inline lib/resourcebuilder/interface.go]: what does `gvk` mean?

**Comment by yifan-gu** [comment]
@abhinavdahiya code wise lgtm, but I'm kinda lost at this point on what's going on with the current cvo state... Maybe need another pair of eyes?

File: lib/resourcebuilder/interface.go (hunk 1 above)
Comment (abhinavdahiya) [reply lib/resourcebuilder/interface.go]: `gvk` : `GroupVersionKind`

**Pull request merged**